homework/index.js

```diff
@@ -28,26 +22,139 @@
+    createAndAppend('td', tr, { text: value });
+    return tr;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, repos) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, {
-          text: err.message,
-          class: 'alert-error',
-        });
-        return;
-      }
-      const ul = createAndAppend('ul', root);
-      repos.forEach(repo => renderRepoDetails(repo, ul));
+  function renderRepoDetails(repo) {
+    const leftSide = document.getElementById('left-side');
+    leftSide.innerHTML = '';
+    const table = createAndAppend('table', leftSide);
+    const tr1 = addRow(table, 'Repository:', '');
+    createAndAppend('a', tr1.lastChild, {
+      href: repo.html_url,
+      text: repo.name,
     });
+    addRow(table, 'Description:', repo.description);
+    addRow(table, 'Fork: ', repo.forks);
+    addRow(table, 'Updated:', formatDate(repo.updated_at));
   }
 
+  function renderContributor(contributor, contributorList) {
+    const li = createAndAppend('li', contributorList);
+    const list = createAndAppend('a', li, {
+      class: 'contributor-item',
+      href: contributor.html_url,
+      target: '_blank',
+    });
+    createAndAppend('img', list, {
+      class: 'image',
+      src: contributor.avatar_url,
+    });
+    createAndAppend('span', list, {
+      class: 'contributor-name',
+      text: contributor.login,
+    });
+    createAndAppend('span', list, {
+      class: 'contribution-count',
+      text: contributor.contributions,
+    });
+    list.addEventListener('click', () => {
+      window.open(contributor.html_url, '_blank');
+    });
+    list.addEventListener('key', event => {
+      if (event.key === 'Enter') window.open(contributor.html_url, '_blank');
+    });
+  }
+  async function fetchAndRender(repo) {
+    try {
+      const contributors = await fetchJSON(repo.contributors_url);
+      const rightSide = document.getElementById('right-side');
+      rightSide.innerHTML = '';
+      createAndAppend('span', rightSide, {
+        text: 'Contributions',
+        class: 'title',
+      });
+      const contributorList = createAndAppend('ul', rightSide, {
+        class: 'contributor-list',
+      });
+      contributors.forEach(contributor =>
+        renderContributor(contributor, contributorList),
+      );
+    } catch (error) {
+      renderError(error);
+    }
+  }
+
+  function startPage(repositories) {
+    const select = document.getElementById('select');
+    repositories.sort((a, b) => a.name.localeCompare(b.name));
+    repositories.forEach((repo, index) => {
+      createAndAppend('option', select, {
+        value: index,
+        text: repo.name,
+      });
+    });
+    renderRepoDetails(repositories[select.value]);
+
+    select.addEventListener('change', () => {
+      renderRepoDetails(repositories[select.value]);
+      fetchAndRender(repositories[select.value]);
+    });
+    return repositories[select.value];
```

Comment:
What's the purpose of returning this value? At which call location is this being used?